### PR TITLE
refactoring/#162 Add support for automated code formatting

### LINKED ISSUE #162: Add support for automated code formatting
## Summary
Add support for automated code formatting using the [black code formatter](https://github.com/psf/black).

## Tasks
- [x] Add black to dev dependencies
- [x] Add configuration section to `pyproject.toml`
   ```
   [tool.black]
   line-length = 88
   verbose = false
   include = "\\.pyi?$"
   ```
- [x] Run initial code formatting on entire code base, [see also](https://black.readthedocs.io/en/stable/guides/introducing_black_to_your_project.html).
- [x] Add/Extend Check target in `noxifle.py`, which automatically will check code formatting 
- [x] Add/Extend Fix target in `noxifle.py`, which automatically will fix code formatting 
- [x] Add CI/CD check(s) if code is properly formatted
- [x] Add `code style` badge to project, see also.

### PR DESCRIPTION
**Pull request by Nicoretti**
fixes #162 

- Add black as dev dependency to the project
- Add basic black configuration
- Add nox target for code formatting
- Migrate code style to Black
- Add new nox target and fix version generator
- Bump version number
- Add version fix to general fix target
- Add Black badge to README.rst

### CODE REVIEW

File: test/test_exadialect_turbodbc.py [hunk 1]
```diff
@@ -4,34 +4,36 @@
...
+            "turbodbc_options"
+        ].read_buffer_size.megabytes == expected_turbodbc.get("read_buffer_size", 50)
+        assert result[
+            "turbodbc_options"
+        ].parameter_sets_to_buffer == expected_turbodbc.get(
+            "parameter_sets_to_buffer", 1000
+        )
+        assert result["turbodbc_options"].use_async_io == expected_turbodbc.get(
+            "use_async_io", False
+        )
```
Comment (tkilias) [inline test/test_exadialect_turbodbc.py:36]: oh, these look weird, I think we should maybe write them differently from the beginning

File: test/test_exadialect_turbodbc.py [hunk 2]
```diff
@@ -41,74 +43,100 @@ def _get_connection_arguments(self, dsn):
...
-        self.assert_parsed("exa+pyodbc://scott:tiger@exa_test", ['exa_test'], expected, {})
+        self.assert_parsed(
+            "exa+pyodbc://scott:tiger@exa_test", ["exa_test"], expected, {}
+        )
 
     def test_create_connect_args_with_turbodbc_args(self):
         expected = self.default_host_args.copy()
         expected.update(self.default_user_args)
         expected.update(DEFAULT_CONNECTION_PARAMS)
-        self.assert_parsed("exa+turbodbc://scott:tiger@192.168.1.2..8:1234/my_schema?"
```
Comment (tkilias) [inline test/test_exadialect_turbodbc.py:80]: What an assert

File: test/test_exadialect_turbodbc.py [hunk 3]
```diff
@@ -41,74 +43,100 @@ def _get_connection_arguments(self, dsn):
...
+        )
 
     def test_create_connect_args_overrides_default(self):
         base_url = "exa+pyodbc://scott:tiger@192.168.1.2..8:1234/my_schema"
 
         before = self._get_connection_arguments(base_url)["inttypesinresultsifpossible"]
-        after = self._get_connection_arguments(base_url + "?inttypesinresultsifpossible=custom")["inttypesinresultsifpossible"]
+        after = self._get_connection_arguments(
+            base_url + "?inttypesinresultsifpossible=custom"
+        )["inttypesinresultsifpossible"]
```
Comment (tkilias) [inline test/test_exadialect_turbodbc.py:141]: Black really discourage this, but it is probably also somewhat a code smell

**Review by tkilias — APPROVED** [review]

File: test/test_exadialect_turbodbc.py (hunk 1 above)
Comment (Nicoretti) [reply test/test_exadialect_turbodbc.py:36]: agreed

**Pull request merged**